src/flint/fmpz.jl

```diff
@@ -1139,19 +1139,22 @@ end
 issquare(x::fmpz) = Bool(ccall((:fmpz_is_square, :libflint), Cint,
                                (Ref{fmpz},), x))
 
+isprime(x::UInt) = Bool(ccall((:n_is_prime, :libflint), Cint, (UInt,), x))
+
 @doc Markdown.doc"""
-    is_prime(x::UInt)
+    isprime(x::fmpz)
 > Return `true` if $x$ is a prime number, otherwise return `false`.
 """
-is_prime(x::UInt) = Bool(ccall((:n_is_prime, :libflint), Cint, (UInt,), x))
+isprime(x::fmpz) = Bool(ccall((:fmpz_is_prime, :libflint), Cint,
+                              (Ref{fmpz},), x))
 
-# flint's fmpz_is_prime doesn't work yet
 @doc Markdown.doc"""
-    isprime(x::fmpz)
+    is_prime(x::Int)
 > Return `true` if $x$ is a prime number, otherwise return `false`.
 """
-isprime(x::fmpz) = Bool(ccall((:fmpz_is_probabprime, :libflint), Cint,
-                              (Ref{fmpz},), x))
+function isprime(n::Int)
+  return isprime(abs(n) % UInt)
```

Comment:
Currently, `isprime` returns `false` for negative numbers, here you make prime some negative numbers...

Reply:
This function is testing if something is prime, and -2 is prime.

If it would be called `isprime_number`, then I agree.

Reply:
> If it would be called isprime_number, then I agree.

The doc says `Return `true` if $x$ is a prime number` though. Anyway I don't mind whether `isprime(-11)` is true, but it's a subtly question which'd better be clarified in the doc, and also be consistent, as here you make `isprime(-11) == true` but `isprime(fmpz(-11)) == false`.

Reply:
You can remove the `abs` call now, as `n` is nonnegative. Also, this is a good candidate for one-line function form (`isprime(n::Int) = n > 0 && isprime(n % UInt)`), which is favored in some style guides, but I don't know if it applies in this project.

Reply:
One line functions are fine. We generally tend to try to keep a "line" under 80 chars for terminals. But of course you will find many exceptions to this, especially as they are often caused by cut'n'paste fixes done over time.

Reply:
The reason is you will get a lot of idiots writing blog posts about how Oscar tells you that -2 is a prime.

Reply:
I understand the rationale. We don't have to discuss or decide it now.